src/main/utils/upload.ts

```diff
@@ -0,0 +1,104 @@
+import path from 'path';
+import * as fs from 'promise-fs';
+import { FILE_STREAM_CHUNK_SIZE } from '../../config';
+import { uploadStatusStore } from '../services/store';
+import { ElectronFile } from '../types';
+
+// https://stackoverflow.com/a/63111390
+export const getFilesFromDir = async (dirPath: string) => {
+    if (!(await fs.stat(dirPath)).isDirectory()) {
+        return [dirPath];
+    }
+
+    let files: string[] = [];
+    const filePaths = await fs.readdir(dirPath);
+
+    for (const filePath of filePaths) {
+        const absolute = path.join(dirPath, filePath);
+        files = files.concat(await getFilesFromDir(absolute));
+    }
+
+    return files;
+};
+
+const getFileStream = async (filePath: string) => {
+    const file = await fs.open(filePath, 'r');
+    let offset = 0;
+    const readableStream = new ReadableStream<Uint8Array>({
+        async pull(controller) {
+            let buff = new Uint8Array(FILE_STREAM_CHUNK_SIZE);
+
+            // original types were not working correctly
```

Comment:
meaning ?

Reply:
The types indicated that the function will return promise of an object of `bytesRead` and `buffer` but it only just `bytesRead`, not sure why.

Reply:
weird... just check if the node version with electron and the @types/node we have in our package JSON are same

Reply:
I tried it and it wasn't same but it didn't solve the issue either.